### DISCUSSION TITLE
ble.sh domain

### DISCUSSION THREAD

**Discussion by KaKi87** [akinomyoga/ble.sh#541] [General]
Hi,

I would like to suggest buying the `ble.sh` domain name to redirect to this repo, like [acmesh-official/acme.sh](https://github.com/acmesh-official/acme.sh) bought `acme.sh` to redirect to their repo.

Thanks

**Comment by akinomyoga** [comment]
If it were free, that would be an appealing idea. However, a three-character domain is expensive. In addition, once we start to use it, we have to continue to pay because it would be converted to a phishing site if it would expire.

**Reply by KaKi87** [reply]
I see you're in Japan, I don't know about the domain providers and currency there, but for me in France, if I would get this, I'd get it from OVH at the cost 60€ per year, which I find really affordable.

**Reply by akinomyoga** [reply]
60€ per year is too expensive for me. If we buy a ten-year plan, it would be 600€.

Although Japanese providers don't seem to cover `.sh`, I think I can technically buy it from a domain provider from the United States or whatever country, and the cost ranges from 40 USD to 80 USD. The cheapest seems 400 USD for ten years.

**Reply by KaKi87** [reply]
How about crowdfunding ?

**Reply by akinomyoga** [reply]
I need to think about the benefits and the maintenance costs. Finally, I likely need to take care of everything. We don't have much content to provide in the domain. Also, I need to think about how long I should maintain it. `ble.sh` is not a project owned and maintained by a big organization, so the day of doom may come depending on my situation.

**Reply by KaKi87** [reply]
What I was thinking was just redirecting https://ble.sh to https://github.com/akinomyoga/ble.sh, nothing more : no maintenance, no additional content.

The initial reason why I had this idea is : when I type `ble.sh` in my browser bar, it's gonna try to load the domain, instead of querying my search engine, just because the `sh` TLD exists, so I have to type again, making sure to manually select the search engine.

Not only would the domain name save time from that error, but it would also save even more time by not having to go through the search engine. That's the benefit.

I feel like the coincidence of you naming the repo with a `.sh` ending while having this TLD in existence is a sign that getting this domain was meant to happen.

**Reply by akinomyoga** [reply]
I tried to search and investigate the crowdfunding platforms, but there seem to be too many platforms. I'm not familiar with them and don't know which one to choose. Each platform seems to have special rules, which might conflict with open-source projects like `ble.sh`. For example, I'm not going to return or offer anything to the supporters, I need to write a project plan, etc. @KaKi87 Could you raise funding in one of some platforms you are familiar with?

**edit**: In case, the cheapest that I mentioned above is [this](https://www.namecheap.com/domains/registration/results/?domain=ble.sh), 385.80 USD / 10 years.
**edit**: [This website](https://porkbun.com/) offers an even cheaper price, 260.60 USD / 10 years, on sale. [ Note: it seems 510.51 USD before discount ]

**Reply by KaKi87** [reply]
For this I would just use [PayPal](https://www.paypal.com/us/digital-wallet/send-receive-money/pool-money) (in fact I have in the past for another domain), and I would pay yearly so I wouldn't have to ask for so much money at once.

**Reply by akinomyoga** [reply]
How is it related to crowdfunding? Or do you mean I need to use my money? I still don't feel it's worth the cost for just redirection. In addition, I need to process it once per year without forgetting it. That would be the maintenance cost.

In addition, I'm not sure if the price would be kept the same for a long time. I'm afraid that they would raise the price when they consider we want to maintain the domain, and they are actually right in the sense that I don't want to let it expire even after they raise the price (to prevent it from being converted to a phishing site).

**Reply by akinomyoga** [reply]
Another possibility would be to prepare a one-time contribution in GitHub sponsors for buying the domain. Then, to collect money until it reaches 260 USD (I might pay a part of it so the value might be decreased). However, a problem is that I'm not sure what to do with the received money when the price is not reached within a year. I need to use the received money within a year because of the tax system in Japan.

**Reply by akinomyoga** [reply]
@KaKi87 Could you pay a part of 260 USD? What would be the maximum amount that you are comfortable with to pay for the domain `ble.sh`?

**Reply by KaKi87** [reply]
> How is it related to crowdfunding?

Look at the link.

> Could you pay a part of 260 USD? What would be the maximum amount that you are comfortable with paying for the domain `ble.sh`?

I don't know.

If you would have agreed to pay yearly, I would have been willing to pay whatever's left to pay when reaching the deadline.

But regarding a 10-year plan, I wouldn't know what to do.

**Reply by akinomyoga** [reply]
> > How is it related to crowdfunding?
> 
> Look at the link.

OK, I haven't checked the link because I thought it would be the top page of PayPal. The explanation seems to imply that I need to ask my friends and family to collect money? It doesn't seem to be "crowd", but is it safe to share the account information online? Or does it expose personal information that should only be shared with your acquaintance?

**Reply by KaKi87** [reply]
> The explanation seems to imply that I need to ask my friends and family to collect money?

There's no such obligation, as I said, I've used the very same feature for the very same purpose.

> is it safe to share the account information online? Or does it expose personal information that should only be shared with your acquaintance?

Just like for personal transactions, it shows your name, which you're already making public (so am I). That's it

**Reply by akinomyoga** [reply]
> > Could you pay a part of 260 USD? What would be the maximum amount that you are comfortable with paying for the domain `ble.sh`?
> 
> I don't know.
> 
> If you would have agreed to pay yearly, I would have been willing to pay whatever's left to pay when reaching the deadline.
> 
> But regarding a 10-year plan, I wouldn't know what to do.

If it would expire in less than ten years, I wouldn't set up the domain in the first place. Once I buy it, I need to continue to pay for it for a long time even after you would disappear. Even if you promise you'd continue to pay every year for a long time of decade order, I'm not comfortable with asking you to pay money to me every year.

A possibility is to buy the ten-year plan at once now by compensating my other budget and collecting money afterward. For example, could you pay 50 USD first? Then, you'll pay "50 USD minus the others' support" every year? If the others' support exceeds 50 USD, that can be used to shorten the period. This way, the collection would finish within four years.

**Reply by akinomyoga** [reply]
Actually, a proper crowdfunding seems easier in regard to the management of the money.

**Reply by KaKi87** [reply]
> If it would expire in less than ten years

Wouldn't you have it renew automatically ?

> I'm not comfortable with asking you to pay money to me every year.

Yet, that's what I'd rather do.

> could you pay 50 USD first? Then, you'll pay "50 USD minus the others' support" every year?

Honestly, if I have to pay that much, I'd rather register the domain name myself, pay it yearly with automated renewal, and get back whatever you and the community would be willing to give.

**Reply by akinomyoga** [reply]
> > If it would expire in less than ten years
> 
> Wouldn't you have it renew automatically ?

I meant I need to continue to pay. I don't think there is a way to renew it automatically without paying...

> > could you pay 50 USD first? Then, you'll pay "50 USD minus the others' support" every year?
> 
> Honestly, if I have to pay that much, I'd rather register the domain name myself, pay it yearly with automated renewal, and get back whatever you and the community would be willing to give.

That works. I actually thought of suggesting it, though one concern was how long you'd maintain it. I don't want to imagine a situation where the domain expires someday and it would start to redirect to a fake GitHub website trying to cheat people.

**Reply by KaKi87** [reply]
> I don't think there is a way to renew it automatically without paying...

Of course not.

> I don't want to imagine a situation where the domain expires someday and it would start to redirect to a fake GitHub website trying to cheat people.

Well, technically, nothing prevents anyone (including myself) to take it purposely for that right now (redirecting it to a clone owned by a similarly named user but shipped with an additional bash function making an HTTP request and executing whatever arbitrary command it returns, imagine that).

So, honestly, I think you should get it just so that no one else could.

As for me :

- I bought a domain for a friend because it was fitting their project's name (`nowplayi.ng`) ;
- I bought a separate domain name for my NAS instead of using a sub of a domain I already own (`kaki87.net`) just because I noticed I could make a pun with my nickname (`kakilo.gy`) ;
- I bought a 4-letter domain name to make an URL shortener that I still haven't made (a satisfying version of) yet, to which I intend to add a free and open subdomain service inspired by `is-a.dev` (`cv.vg`) ;
- My brother and I bought a very very expensive (still not fully paid yet) domain name for a company we're building (`friendlyapps.com`).

I hope this shines a light on my interest in domain names :)

**Reply by akinomyoga** [reply]
> Well, technically, nothing prevents anyone (including myself) to take it purposely for that right now

That's not the problem. Currently, no people know that "`https://ble.sh` is an official and trustable one". The problem happens when the web page is suddenly replaced after people learn that `https://ble.sh` is an official one and start to trust the redirected page without paying attention. Exactly for the same reason, it is unlikely to happen now because there is less incentive for malicious people to buy `ble.sh` for cheating now.

**Reply by KaKi87** [reply]
> it is unlikely to happen now because there is less incentive for malicious people to buy `ble.sh` for cheating now.

It will become likely to happen whenever your projects gets further traction. That's why there are fake VLC websites, fake LibreOffice websites, etc. It doesn't matter that the website was never certified as legitimate, as long as there are people who will stumble on it from their browser bar and believe it's the real thing.

**Reply by akinomyoga** [reply]
> > it is unlikely to happen now because there is less incentive for malicious people to buy `ble.sh` for cheating now.
> 
> It will become likely to happen whenever your projects gets further traction.

I wrote "now"; `ble.sh` is not as popular now. Also, I didn't write there is no possibility. I agree that there is still a non-zero probability that malicious people buy it now, but it's lower by an order of magnitude than the probability for an expired domain, where the malicious people can exploit the belief of the users who got used to typing `https://ble.sh` to visit the official one.

`ble.sh` is not as popular as VLC and LibreOffice (if you mean VLC player by VLC) now, and I don't think it would become as popular as them. There may be still a possibility that `ble.sh` will become more popular than now (even if it wouldn't compare with VLC and LibreOffice), but if we start to care about them, one cannot make any scripts ending with `.sh` public, or everyone needs to buy the corresponding domain in advance. That's impractical. Even if one buys one domain, there are many other variations in the domain name. For this reason, there are already many fake websites, and internet users are supposed to know that a website you found by randomly typing a URL in the address bar is not trustworthy. People instead rely on the search engine results, their memory about the URL, etc., where the situation with a hijacked expired domain is critical.

**Reply by KaKi87** [reply]
All true indeed.

**Reply by akinomyoga** [reply]
So I'd like to consider the planning for the ten-year maintenance before buying the domain. I'll have to again consider what to do after ten years, but I'll take care of it. There are still options.

- I'm not familiar with crowdfunding, but I still feel that crowdfunding seems simpler. It covers the case where the required price is not reached. If it doesn't reach, I guess the money will be automatically returned to the supporters. A non-trivial factor is whether a sufficient number of people support buying the domain.
- Or I might create a one-time contribution in GitHub sponsors and collect money for the domain. Then, if the value reaches the required price for the cheapest ten-year plan, we buy the domain at that point. Unlike crowdfunding, we can collect money for a longer time. I can start it the next January 1st. If it doesn't reach the price by Dec 31st, 2025, I'll have to pay an extra tax, but I'll cover it.
- Or you can raise funding by your PayPal account and collect money, since you are familiar with it. Then, you or I can buy the domain when it reaches the price.
- Or you can actually maintain the domain `ble.sh` (with yearly renewing) by yourself by collecting money by whatever means, if you are responsible with maintaining the domain for ten years (or finding another maintainer if it becomes difficult for you).

**Reply by KaKi87** [reply]
Alright, keep me posted.

**Reply by KaKi87** [reply]
Well, less than a month later, an individual bought the domain name just to use as a personal homepage 😭 (that doesn't even link to their contacts and socials, and that replaces a longer domain name they already owned)

**Reply by akinomyoga** [reply]
Yeah, I noticed that soon after the domain was taken. Maybe we shouldn't have talked about the domain purchase in public.

**Reply by KaKi87** [reply]
You think the person did it on purpose ? I mean, the domain does matches their initials, and it has real content. I'd think it would have happened anyways :thinking: 

In fact, I wonder if they'd be okay with adding a small footer link from their page to your repo, if I could find their socials.

**Reply by akinomyoga** [reply]
I'm not sure if it was on purpose, but there is a possibility.

**Reply by KaKi87** [reply]
Found only this so far : https://x.com/brdnle (I don't have Twitter)
